proxy-libs/heliosmongo/mongo.go

```diff
@@ -0,0 +1,164 @@
+type ChangeStream = originalMongo.ChangeStream
+type Client = originalMongo.Client
+type ClientEncryption = originalMongo.ClientEncryption
+type Collection = originalMongo.Collection
+type CollectionSpecification = originalMongo.CollectionSpecification
+type CommandError = originalMongo.CommandError
+type Cursor = originalMongo.Cursor
+type Database = originalMongo.Database
+type DatabaseSpecification = originalMongo.DatabaseSpecification
+type DeleteManyModel = originalMongo.DeleteManyModel
+type DeleteOneModel = originalMongo.DeleteOneModel
+type DeleteResult = originalMongo.DeleteResult
+type Dialer = originalMongo.Dialer
+type EncryptionKeyVaultError = originalMongo.EncryptionKeyVaultError
+type ErrMapForOrderedArgument = originalMongo.ErrMapForOrderedArgument
+type IndexModel = originalMongo.IndexModel
+type IndexOptionsBuilder = originalMongo.IndexOptionsBuilder
+type IndexSpecification = originalMongo.IndexSpecification
+type IndexView = originalMongo.IndexView
+type InsertManyResult = originalMongo.InsertManyResult
+type InsertOneModel = originalMongo.InsertOneModel
+type InsertOneResult = originalMongo.InsertOneResult
+type ListDatabasesResult = originalMongo.ListDatabasesResult
+type MarshalError = originalMongo.MarshalError
+type MongocryptError = originalMongo.MongocryptError
+type MongocryptdError = originalMongo.MongocryptdError
+type Pipeline = originalMongo.Pipeline
+type ReplaceOneModel = originalMongo.ReplaceOneModel
+type RewrapManyDataKeyResult = originalMongo.RewrapManyDataKeyResult
+type ServerError = originalMongo.ServerError
+type Session = originalMongo.Session
+type SessionContext = originalMongo.SessionContext
+type SingleResult = originalMongo.SingleResult
+type StreamType = originalMongo.StreamType
+type UpdateManyModel = originalMongo.UpdateManyModel
+type UpdateOneModel = originalMongo.UpdateOneModel
+type UpdateResult = originalMongo.UpdateResult
+type WriteConcernError = originalMongo.WriteConcernError
+type WriteError = originalMongo.WriteError
+type WriteErrors = originalMongo.WriteErrors
+type WriteException = originalMongo.WriteException
+type WriteModel = originalMongo.WriteModel
+type XSession = originalMongo.XSession
+
+var ErrClientDisconnected = originalMongo.ErrClientDisconnected
+var ErrEmptySlice = originalMongo.ErrEmptySlice
+var ErrInvalidIndexValue = originalMongo.ErrInvalidIndexValue
+var ErrMissingResumeToken = originalMongo.ErrMissingResumeToken
+var ErrMultipleIndexDrop = originalMongo.ErrMultipleIndexDrop
+var ErrNilCursor = originalMongo.ErrNilCursor
+var ErrNilDocument = originalMongo.ErrNilDocument
+var ErrNilValue = originalMongo.ErrNilValue
+var ErrNoDocuments = originalMongo.ErrNoDocuments
+var ErrNonStringIndexName = originalMongo.ErrNonStringIndexName
+var ErrUnacknowledgedWrite = originalMongo.ErrUnacknowledgedWrite
+var ErrWrongClient = originalMongo.ErrWrongClient
+
+func setMonitor(opts []*options.ClientOptions) []*options.ClientOptions {
+	monitor := otelmongo.NewMonitor()
```

Comment:
We add it to each 'opt' - is that necessary?

Reply:
In this example - https://github.com/open-telemetry/opentelemetry-go-contrib/blob/9fd4fa889a38e38f73c10d4872bf4a64e275f157/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/example_test.go#L32 - only one client has it.

Reply:
Yes, but notice that `Connect` and `NewClient` receive an array of options (`...opts`), so the `Monitor` has to be in all of them (the library merges all the options in the array, so to be sure that the `Monitor` is set, I push it to all the opts.